File: Easy/2026-08-14-3090-Maximum-Length-Substring-With-Two-Occurrences/solution.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int left = 0, right = 0, maxLength = 0;
        int[] count = new int[26];

        while (right < s.length()) {
            count[s.charAt(right) - 'a']++;
            
            while (count[s.charAt(right) - 'a'] > 2) {
                count[s.charAt(left) - 'a']--;
                left++;
            }
            
            maxLength = Math.max(maxLength, right - left + 1);
            right++;
        }
        
        return maxLength;
    }
}

File: Easy/2026-08-14-3090-Maximum-Length-Substring-With-Two-Occurrences/solution.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        left = 0
        count = {}
        max_length = 0
        
        for right in range(len(s)):
            count[s[right]] = count.get(s[right], 0) + 1
            
            while any(v > 2 for v in count.values()):
                count[s[left]] -= 1
                if count[s[left]] == 0:
                    del count[s[left]]
                left += 1
            
            max_length = max(max_length, right - left + 1)
        
        return max_length